docs/smart-contracts/modules.md

```diff
@@ -0,0 +1,564 @@
+In a typical smart contract—the full application—each individual module can provide a focused set of functionality with clear organizational logic.
+
+Because a smart contract can be defined using one module or many modules, the logic in individual Pact files (`.pact`) is always referred to as a module.
+
+## Module declaration
+
+You can create a module by typing the `module` keyword, followed by the module name and the keyset or governance capability that owns the module.
+The following example illustrates a module named **example** that is governed by the **admin-keyset** referenced in the first line of the module declaration:
+
+```pact
+;; Define and read keysets
+
+(module example "admin-keyset"
+    ;; module code goes here
+
+    ;; function calls
+)
+```
+
+The entire **example** module consists of the code within the opening and closing parentheses. 
+Top-level code, like `namespace` definitions and `keyset` definitions, are outside of these parentheses.
+For example, you enter a namespace and define a keyset for a module before starting the module declaration like this:
+
+```pact
+;; Enter a namespace
+(namespace "free")
+
+;; Define and read a keyset
+(define-keyset "free.admin-keyset" (read-keyset "admin-keyset"))
+
+(module example "admin-keyset"
+    ;; module code goes here
+
+    ;; function calls
+)
+```
+
+## Module governance
+
+With keyset governance like the previous example, the `admin-keyset` is defined outside of the module and checked and enforced at the module level. 
+Any attempt to upgrade the module, write to module tables, or access table functions directly requires the `admin-keyset` to sign the transaction.
+As an alternative to strict keyset enforcement, you can specify a _module governance_ capability in the module declaration to support a more generalized form of module governance. 
+By using a governance capability that references a `defcap` declaration in the module body, you can define more flexible models for enforcing access to Pact modules, tables, and functions. 
+
+For example, you can implement the same governance for the **example** module using a governance capability named **GOVERNANCE** like this:
+
+```pact
+(module example GOVERNANCE
+  ...
+  (defcap GOVERNANCE ()
+    (enforce-keyset "admin-keyset"))
+  ...
+)
+```
+
+Note that the capability name has no significance, except to indicate the purpose of the capability. 
+Its placement at the beginning of the module declaration is what identifies this capability as a module governance capability.
+
+It's worth noting that, when you initially deploy a module, the module governance capability is not invoked. 
+This behavior is different than when you use use a keyset.
+With a keyset, the keyset must always be defined and evaluated to ensure that the keyset exists before a module can be deployed.
+Therefore, you might consider module governance using a capability to be more risky than using a keyset. 
```

Comment:
I don't feel this is a correct statement. @emilypi can you comment?

Reply:
Where this came from:
https://docs.kadena.io/build/pact/advanced#generalized-module-governanceh-2066495538

### Capability risks

Also, this means that, when initially installing a module, the governance function is not invoked. This is different behavior than when a keyset is specified: the keyset must be defined and it is enforced, to ensure that the keyset actually exists.

Module governance is therefore more "risky" as it can mean that the module cannot be upgraded if there is a bug in the governance capability. Clearly, care must be taken when implementing module capabilities, and using the Pact formal verification system is highly recommended here.